android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java

```diff
@@ -525,6 +572,86 @@ public void onError(String error) {
         });
     }
 
+    @ReactMethod
+    public void suspendOfflinePack(final String packName, final Promise promise) {
+        mainHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                OfflineRegionProgressObserver foundObserver = null;
+
+                for (OfflineRegionProgressObserver observer : offlinePackObservers) {
+                    if (packName.equals(observer.name)) {
+                        foundObserver = observer;
+                        break;
+                    }
+                }
+
+                if (foundObserver == null) {
+                    promise.resolve(Arguments.createMap());
+                    return;
+                }
+
+                foundObserver.region.setDownloadState(OfflineRegion.STATE_INACTIVE);
+                final OfflineRegionProgressObserver _foundObserver = foundObserver;
+                foundObserver.region.getStatus(new OfflineRegion.OfflineRegionStatusCallback() {
+                    @Override
+                    public void onStatus(OfflineRegionStatus status) {
+                        //_foundObserver.fireUpdateEvent();
+                        _foundObserver.onStatusChanged(status);
+                        WritableMap result = Arguments.createMap();
+                        result.putString("suspended", _foundObserver.name);
+                        promise.resolve(result);
+                    }
+                    @Override
+                    public void onError(String error) {
+                        Log.e(context.getApplicationContext().getPackageName(), error);
+                        promise.reject(new JSApplicationIllegalArgumentException(error));
+                    }
+                });
+            }
+        });
+    }
+
+    @ReactMethod
+    public void resumeOfflinePack(final String packName, final Promise promise) {
+        mainHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                OfflineRegionProgressObserver foundObserver = null;
+
+                for (OfflineRegionProgressObserver observer : offlinePackObservers) {
```

Comment:
We could break out finding the observer into it's own method like `private OfflineRegionProgressObserver getObserver(String name)` and we can then initialize the variable as `final OfflineRegionProgressObserver foundObserver` instead of having to initialize `foundObserver` and `_foundObserver`. Will also help with reducing code